client/src/common/features.ts

```diff
@@ -552,6 +552,11 @@ export abstract class TextDocumentLanguageFeature<PO, RO extends TextDocumentReg
 		if (!documentSelector || !capability) {
 			return undefined;
 		}
+
+		if (documentSelector && capability) {
```

Comment:
We should only log something if the document selectors are different. Specifying capabilities is fine since there are no client default once. In addition we should somehow log for which feature this is happening.

Reply:
Since both `documentSelector` and `capability` are objects, I ended up going with comparing their JSON stringified versions. I don't really like that approach, so let me know if you have a preferred way of comparing the selectors.

There are a few possible types for selectors. If you prefer, we can definitely add checks for each possible combination.

I also added the method name through `this._registrationType.method`.